client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -160,19 +162,41 @@ private Checkpoint completeCheckpoint(String checkpointName, StateSynchronizer<R
     }
 
     @Override
-    public void resetReadersToCheckpoint(Checkpoint checkpoint) {
+    public void resetReaders(final Checkpoint checkpoint) {
+        resetReaders(checkpoint.asImpl().getPositions());
+    }
+
+    @Override
+    public void resetReaders(final Map<Stream, StreamCut> streamCuts) {
         @Cleanup
         StateSynchronizer<ReaderGroupState> synchronizer = createSynchronizer();
+        synchronizer.fetchUpdates();
+
+        //ensure that streamCut for all the streams managed by the ReaderGroup are present.
+        Preconditions.checkArgument(validateStreamCuts(streamCuts.values(), synchronizer.getState().getStreamNames()),
+                "StreamCuts for streams managed by the ReaderGroup need to be provided");
+
+        //reset the Readers to the StreamCut.
         synchronizer.updateState(state -> {
             ReaderGroupConfig config = state.getConfig();
             Map<Segment, Long> positions = new HashMap<>();
-            for (StreamCut cut : checkpoint.asImpl().getPositions().values()) {
+            for (StreamCut cut : streamCuts.values()) {
                 positions.putAll(cut.getPositions());
             }
             return Collections.singletonList(new ReaderGroupStateInit(config, positions));
         });
     }
 
+    /*
+        Validate the following
+            - provided stream cuts are unique and has no duplicates.
+            - stream cuts for all the streams managed by the reader group are present.
```

Comment:
It might be confusing, but I want to raise the possibility of a cut including only a strict subset of the streams of the reader group. Say for example that I take a cut C1, reconfigure the reader group to include another stream, and now I want to use C1. I think this scenario is forbidden with this API, and I wonder if this is something we would like to support. The bevavior could be that the reader group start the missing streams from offset zero.

Reply:
Will handle this as part of fixing #2326